src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -3391,19 +3415,34 @@ public class JITStats
         /// Update method statistics
         /// </summary>
         /// <param name="_method"></param>
-        internal void Update(TraceJittedMethod _method)
+        public void Update(TraceJittedMethod _method)
```

Comment:
Why the switch to public?  The challenge with this method as written (current situation) is that it must be called exactly once per method - if called more than once it will result in over counting.  As a tradeoff we kept this internal so that we had more control over the callers.

Reply:
In JitStatsEx.Create PerfView is rebuilding per-module filtered JitStats instances. Previously to construct these instances it had a duplicate copy-pasted implementation of Update() inlined in the for loops. I changed it to call Update() rather than duplicate all my logic changes in both places, mostly for expediency as I tinkered. I could restore the copy-paste style cloning, or perhaps a cleaner option you'd prefer is creating a constructor that takes as input a set of TraceJittedMethod and outputs a JitStats instance?

Reply:
I see.... between a rock and hard place.  I agree duplicating code is less than ideal.  I think a little more work is warranted to avoid the easy mistake of calling this internal implementation detail in the wrong situation which will result in overcounting.  The constructor solution seems reasonable, if that works for your perfview.

Reply:
Hmm, after thinking about various options I have concluded that the current design, where JITStats is basically simply a 'container of fields' and is WRITABLE is useful (it allows users to make any kinds of statitics over JIT methods that they want).   As such it is REASONABLE to have a public mutator that updates the statistics given a method (which is what UPDATE does) because that is HOW YOU USE THE CLASS (you accumulate the particular methods you want into the stats class).  

Thus I think it is OK to expose the Update() method, but I would like it renamed to be 'AddMethodToStatistics' when we make it public.  

Unless Jeff hates this idea, lets go with that...

Reply:
lgtm